c7n/resources/redshift.py

```diff
@@ -141,7 +141,13 @@ def process_db_set(self, db_set):
             else:
                 params['FinalClusterSnapshotIdentifier'] = snapshot_identifier(
                     'Final', db['ClusterIdentifier'])
-            c.delete_cluster(**params)
+            try:
+                c.delete_cluster(**params)
+            except ClientError as e:
+                if e.response['Error']['Code'] == "InvalidClusterState":
+                    self.log.warning("Exception deleting cluster: %s" %e.response['Error']['Code'])
```

Comment:
after looking again, I updated the wording to immediately indicate the issue, and I added the clusteridentifier as well.